src/init.c

```diff
@@ -292,6 +304,29 @@ shmem_internal_init(int tl_requested, int *tl_provided)
               shmem_internal_heap_base, shmem_internal_heap_length,
               shmem_internal_data_base, shmem_internal_data_length);
 
+#ifdef HAVE_SCHED_GETAFFINITY
+    if (shmem_internal_params.DEBUG) {
+        CPU_ZERO(&my_set);
+
+        ret = sched_getaffinity(0, sizeof(my_set), &my_set);
+
+        if (ret == 0) {
+            char *cores_str = malloc(sizeof(char) * CPU_SETSIZE * 5 + 2);
+            strcpy(cores_str," ");
+            for (int i = 0; i < CPU_SETSIZE; i++) {
+                if (CPU_ISSET(i, &my_set)) {
+                    core_count++;
+                    char cpunum[5];
+                    sprintf(cpunum, "%d ", i);
+                    strcat(cores_str, cpunum);
+                }
+            }
```

Comment:
I think a structure like we use in `DEBUG_MSG` where we `snprintf` directly into the string buffer might be better.  The `strcat` call is O(N) since it needs to seek to the end of the string each time (not a huge deal, but the intermediate string isn't saving any time).  Something like this avoids seeking each time and extra copies:

```
size_t off = 1; /* start after " " */
for (int i = 0; i < CPU_SETSIZE; i++) {
    if ... {
        ...
        off = snprintf(cores_str+off, CPU_SETSIZE*5+2-off, "%d ", i);
        ...
    }
...
}